src/SonarScanner.MSBuild.Shim/SonarScanner.Wrapper.cs

```diff
@@ -140,9 +141,19 @@ private static void IgnoreSonarScannerHome(ILogger logger)
     /// <summary>
     /// Returns any additional environment variables that need to be passed to the sonar-scanner.
     /// </summary>
-    private static IDictionary<string, string> GetAdditionalEnvVariables(ILogger logger)
+    private static IDictionary<string, string> GetAdditionalEnvVariables(AnalysisConfig config, ILogger logger)
     {
         IDictionary<string, string> envVarsDictionary = new Dictionary<string, string>();
+        if (!string.IsNullOrWhiteSpace(config.JavaExePath))
+        {
+            // The java exe path points to the java.exe file while the JAVA_HOME needs to point to the installation directory that contains the bin/ directory where the java executable
+            // physically resides.
+            // e.g. C:\Program Files\Java\jdk-17\bin\java.exe -> C:\Program Files\Java\jdk-17\
+            var exeDirectory = Path.GetDirectoryName(config.JavaExePath);
+            var javaHome = Directory.GetParent(exeDirectory).ToString();
```

Comment:
Both `GetDirectoryName` and `GetParent` can throw.
Should we add a try-catch and log the error?